### PR TITLE
Introduce DualView sample

### PR DESCRIPTION
**Pull request by joyl1216**
Screenshot_1612397556 copy
Screenshot_1612397550 copy
Screenshot_1612402836 copy
Screenshot_1612397564 copy

### CODE REVIEW

File: DualView/.gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
+*.iml
```
Comment (CesarValiente) [inline DualView/.gitignore:1]: I think this is a better `.gitignore` file: https://github.com/microsoft/surface-duo-compose-samples/blob/main/ExtendedCanvas/.gitignore

File: DualView/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *  * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  * Licensed under the MIT License.
+ *
+ */
+
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdkVersion 30
```
Comment (CesarValiente) [inline DualView/app/build.gradle:12]: not sure if in other modules these values are placed also in a `dependencies.gradle` file. Please, consider it.

File: DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    private lateinit var windowManager: WindowManager
+    private lateinit var appStateViewModel: AppStateViewModel
+
+    private val handler = Handler(Looper.getMainLooper())
+    private val mainThreadExecutor = Executor { r: Runnable -> handler.post(r) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        windowManager = WindowManager(this)
+        appStateViewModel = ViewModelProvider(this).get(AppStateViewModel::class.java)
+        appStateViewModel.setSelectionLiveData(-1)
```
Comment (CesarValiente) [inline DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/MainActivity.kt:33]: please, don't use magic numbers, better to have it in a constant with a meaningful name, so we know what that number represents.

File: DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/MapView.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.res.imageResource
+import com.microsoft.device.display.samples.dualview.models.AppStateViewModel
+import com.microsoft.device.display.samples.dualview.models.restaurants
+
+@Composable
+fun MapView(modifier: Modifier, appStateViewModel: AppStateViewModel) {
+    val selectionLiveData = appStateViewModel.getSelectionLiveData()
+    val selectedIndex = selectionLiveData.observeAsState(initial = -1).value
```
Comment (CesarValiente) [inline DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/MapView.kt:35]: The same here with the `-1`, better to have it in a constant so we know what it is and we can reuse it (as it is used after that).

File: DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+import com.microsoft.device.display.samples.contentcontext.util.formatPriceRange
+import com.microsoft.device.display.samples.contentcontext.util.formatRating
+import com.microsoft.device.display.samples.dualview.models.AppStateViewModel
+import com.microsoft.device.display.samples.dualview.models.Restaurant
+import com.microsoft.device.display.samples.dualview.models.restaurants
+import com.microsoft.device.display.samples.dualview.ui.theme.selectedBody1
+import com.microsoft.device.display.samples.dualview.ui.theme.selectedBody2
+import com.microsoft.device.display.samples.dualview.ui.theme.typography
+
+private val outlinePadding = 25.dp
```
Comment (CesarValiente) [inline DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt:45]: is this using an extension function to convert it to dps?

Comment (joyl1216) [reply DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt:45]: No, why do we need a extension function?

Comment (CesarValiente) [reply DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt:45]: No, I mean, is `.dp` an extension function?

Comment (joyl1216) [reply DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt:45]: No, it is just how the unit is used in Jetpack Compose. https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/package-summary#dp_2

Comment (CesarValiente) [reply DualView/app/src/main/java/com/microsoft/device/display/samples/dualview/RestaurantView.kt:45]: Ah oki, so they are extension functions created for and in the compose framework. Cool!

File: ComposeGallery/dependencies.gradle [hunk 6]
```diff
@@ -30,8 +30,8 @@ ext {
     viewPager2Version = "1.0.0"
     localBroadcastManagerVersion = "1.0.0"
     preferenceVersion = "1.1.1"
-    composeVersion = "1.0.0-alpha10"
-    windowVersion = "1.0.0-alpha01"
+    composeVersion = "1.0.0-alpha11"
+    windowVersion = "1.0.0-alpha02"
```
Comment (CesarValiente) [inline ComposeGallery/dependencies.gradle:34]: nice one!

**Review by CesarValiente — APPROVED** [review]
LGTM! Thanks Joy!

**Pull request merged**